### PR TITLE
Complete vld1 instructions with some corrections

### PR DESCRIPTION
**Pull request by SparrowLii**
This PR does the following:
(1) completes` vld1` neon instructions, mainly for `*_p64`s
(2) replaces `crypto` feature with `aes` for arm 
(3) adds `"core_arch/src/arm_shared/neon"` dir in `stdarch-verify`. Based on this, the function names of `vmull_n_s16`, `vmull_n_s32`, `vmull_n_u16`, `vmull_n_u32`, `vqdmulhq_n_s16`, `vqdmulhq_n_s32` have been corrected
(4) optimizes `stdarch-gen` so that `vld2*` (and other) instructions can be generated more clearly(For the convenience of reviewing, they are not generated in this PR)

### CODE REVIEW

File: crates/stdarch-gen/neon.spec [hunk 1]
```diff
@@ -2429,8 +2594,8 @@ aarch64 = pmull
 generate poly64x2_t:poly64x2_t:p128
 
 /// Vector long multiply with scalar
-name = vmull
-n-suffix
+name = vmull_n
+no-q
```
Comment (hkratz) [inline crates/stdarch-gen/neon.spec:2598]: Maybe this change should be documented in the PR or in an extra commit as they change the names of public intrinsics (to their correct names).

File: crates/stdarch-gen/neon.spec [hunk 2]
```diff
@@ -3568,7 +3733,7 @@ generate int16x4_t:i16:int16x4_t, int32x2_t:i32:int32x2_t
 
 /// Vector saturating doubling multiply high with scalar
 name = vqdmulhq_n
-out-suffix
+no-q
```
Comment (hkratz) [inline crates/stdarch-gen/neon.spec:3736]: See above.

File: crates/stdarch-gen/neon.spec (hunk 1 above)
Comment (SparrowLii) [reply crates/stdarch-gen/neon.spec:2598]: Thanks for pointing it out! I added relevant documents to the description of the PR.

File: crates/core_arch/src/aarch64/neon/mod.rs [hunk 3]
```diff
@@ -749,6 +749,22 @@ pub unsafe fn vld1q_p16(ptr: *const p16) -> poly16x8_t {
...
+pub unsafe fn vld1_p64(ptr: *const p64) -> poly64x1_t {
+    transmute(u64x1::new(*ptr))
+}
+
+/// Load multiple single-element structures to one, two, three, or four registers.
+#[inline]
+#[target_feature(enable = "neon,aes")]
+#[cfg_attr(test, assert_instr(ldr))]
+pub unsafe fn vld1q_p64(ptr: *const p64) -> poly64x2_t {
+    transmute(u64x2::new(*ptr, *ptr.offset(1)))
```
Comment (Amanieu) [inline crates/core_arch/src/aarch64/neon/mod.rs:765]: There was a concern raised in #1148 that this doesn't get optimized down to a single instruction.

Comment (hkratz) [reply crates/core_arch/src/aarch64/neon/mod.rs:765]: I agree, all others have been already changed to use `read_unaligned()` instead which cannot be seen in the diff as it this PR is based on an older commit.

Comment (SparrowLii) [reply crates/core_arch/src/aarch64/neon/mod.rs:765]: That makes sense.

**Comment by hkratz** [comment]
@Amanieu Can this be merged? Fixing #1212 and #1217 requires changes to stdarch-gen and this PR contains a big refactoring of it. I would rather like to build on this instead of running into rebase conflicts.

**Pull request merged**